src/osd/OSDMap.h

```diff
@@ -174,6 +174,10 @@ class OSDMap {
     float new_full_ratio = -1;
 
     int8_t new_require_min_compat_client = -1;
+    vector<int32_t> new_noout_osds;
+    vector<int32_t> old_noout_osds;
+    vector<int32_t> new_nodown_osds;
+    vector<int32_t> old_nodown_osds;
```

Comment:
I think the way to do this is actually with the osd_state bits... except that we're out of bits.  I can do a quick PR to expand it to a uint32_t, though, with the switch to luminous.  Then you can base this PR on top of that.